docs/design/flow-monitoring/README.md

```diff
@@ -17,13 +17,26 @@ ISL cache structure:
 ISL | last RTT latency | RTT latency expiration time | last one way latency
 --- | --- | --- | ---
 
-![Flow-monitoring](flow-monitoring.png "Flow monitoring")
+![Flow-monitoring-topology](flow-monitoring-topology.png "Flow monitoring")
 
-On the topology startup both caches initialized with data from the database. During the topology lifetime system events helps to keep caches in actual states. `flow-hs` will notify `flow-monitor` about any flow path changes using additional events. All flow-modifying FSMs will execute `NOTIFY_FLOW_MONITOR` action before the finish. 
+On the topology startup both caches initialized with data from the database. During the topology lifetime system events helps to keep caches in actual states. `flow-hs` will notify `flow-monitor` about any flow path changes using additional events. All flow-modifying FSMs will execute `NOTIFY_FLOW_MONITOR` action before the finish. `network` topology will notify `flow-monitor` about ISL changes.
 
 ![Isl-cache-update](isl-cache-update.png "Isl cache update")
 ![Flow-cache-update](flow-cache-update.png "Flow cache update")
+![Flow-latency-check](flow-latency-check.png "Flow latency check")
 
 ## Flow SLA reactions
 
-To be discussed and documented in the future.
\ No newline at end of file
+Every calculated flow latency value will be sent to stats topology and saved in openTSDB.
+
+Flow SLA reactions are applied only to flows with latency-based path computation strategies. Flow has two levels of latency SLA called max_latency and max_latency_tier_2. When actual flow latency is lower than max_latency flow is considered as UP, when latency is in between max_latency and max_latency_tier_2 flow is considered as DEGRADED, in other case flow is considered as DOWN. Threshold and timeout is used to prevent flow status flapping when latency is fluctuating around one of the SLA levels. Flow latency should exceed SLA level +/- threshold or stay stable for a timeout time window before system will change actual flow status. 
```

Comment:
When we have flow with latency close to it's SLA some measurements may slightly violate exact SLA level due to measurement deviation. We want to apply some actions in two cases: 
1. when we have such small latency burst for a long period of time; 
2. when latency measurement greatly exceeds SLA.
The first part is covered by timeout and the second by threshold.

Reply:
So in the second case, we do not filter sporadic violates with timeout value ie a single spike above SLA + offset will trigger action? If it is so, it looks incorrect, because of mentioned measurement deviations. 

If not, it means that the action will be triggered:
* if latency violates SLA + offset and stays above this line for timeout seconds
* it latency violates SLA and stays above this line for timeout seconds

i.e. the conditions are the same, except SLA level - in one case we have some offset in others we have no offset.

PS On the thresholds graph you have 2 offset above and below, i.e positive and negative I can't understand the meaning of negative offset.

Reply:
Reworked and removed threshold.

Reply:
Ok. I can be wrong - without "thresholds" (abow and below) SLA value we can have a case when latency jups above and below SLA on each consequented measurement and system will not react on this behaviour.

So we can treat SLA boundary as 2 boundaries moved above and below this boundary on some offset (it can 2 two different offset), lets name tham SLA+ and SLA-. So we will register SLA vioulation if our measuremnts shows value above SLA+ - monitor should trigger `tier1fail` event. If latency value become below SLA- boundary, monitor will trigger `healthy` event.

Reply:
Reworked and moved thresholds back.